test/ValueGetterTestHelpers.h

```diff
@@ -0,0 +1,290 @@
+//  Copyright 2025, University of Freiburg,
+//  Chair of Algorithms and Data Structures.
+//  Authors: @DuDaAG,
+//           Christoph Ullinger <ullingec@cs.uni-freiburg.de>
+
+#ifndef QLEVER_TEST_VALUEGETTERTESTHELPERS_H
+#define QLEVER_TEST_VALUEGETTERTESTHELPERS_H
+
+#include <gtest/gtest.h>
+
+#include "./GeometryInfoTestHelpers.h"
+#include "./SparqlExpressionTestHelpers.h"
+#include "engine/LocalVocab.h"
+#include "engine/sparqlExpressions/SparqlExpressionValueGetters.h"
+#include "global/Constants.h"
+#include "index/LocalVocabEntry.h"
+#include "parser/Literal.h"
+#include "parser/LiteralOrIri.h"
+#include "util/GeometryInfo.h"
+
+namespace valueGetterTestHelpers {
+
+const std::string ttl = R"(
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+<x> <y> "anXsdString"^^xsd:string, 
+        "someType"^^<someType>,
+        "noType".
+  )";
+struct TestContextWithGivenTTl {
+  std::string turtleInput;
+  QueryExecutionContext* qec = ad_utility::testing::getQec(turtleInput);
+  VariableToColumnMap varToColMap;
+  LocalVocab localVocab;
+  IdTable table{qec->getAllocator()};
+  sparqlExpression::EvaluationContext context{
+      *qec,
+      varToColMap,
+      table,
+      qec->getAllocator(),
+      localVocab,
+      std::make_shared<ad_utility::CancellationHandle<>>(),
+      sparqlExpression::EvaluationContext::TimePoint::max()};
+  std::function<Id(const std::string&)> getId =
+      ad_utility::testing::makeGetId(qec->getIndex());
+  TestContextWithGivenTTl(std::string turtle)
+      : turtleInput{std::move(turtle)} {}
+};
+
+// Helper function to check literal value and datatype
+inline void checkLiteralContentAndDatatype(
+    const std::optional<ad_utility::triple_component::Literal>& literal,
+    const std::optional<std::string>& expectedContent,
+    const std::optional<std::string>& expectedDatatype) {
+  if (literal.has_value()) {
+    ASSERT_EQ(asStringViewUnsafe(literal.value().getContent()),
+              expectedContent.value_or(""));
+
+    if (literal.value().hasDatatype()) {
+      ASSERT_TRUE(expectedDatatype.has_value());
+      ASSERT_EQ(asStringViewUnsafe(literal.value().getDatatype()),
+                expectedDatatype.value());
+    } else {
+      ASSERT_FALSE(expectedDatatype.has_value());
+    }
+  } else {
+    ASSERT_FALSE(expectedContent.has_value());
+  }
```

Comment:
Same here, you didn't write this code, but it seems an awful lot like an equality check of two `triple_component::Literal` types, but more complicated.